sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala

```diff
@@ -93,6 +93,22 @@ class ResolveCatalogs(val catalogManager: CatalogManager)
         s"Can not specify catalog `${catalog.name}` for view ${tableName.quoted} " +
           s"because view support in catalog has not been implemented yet")
 
+    case RenameTableStatement(NonSessionCatalog(catalog, oldName), newNameParts, isView) =>
+      if (isView) {
+        throw new AnalysisException("Renaming view is not supported in v2 catalogs.")
+      }
+      newNameParts match {
+        case NonSessionCatalog(newCatalog, newName) =>
```

Comment:
If there is a name conflict, this forces users to always specify the catalog name.

I think it's easier to define the `newNameParts` as the identifier of the target catalog, so that we can just return
`RenameTable(atalog.asTableCatalog, oldName.asIdentifier, newNameParts.asIdentifier)`

Reply:
Thanks for the suggestion, but wanted to make sure if I understood correctly.

Are you suggesting `newNameParts` will contain only the identifier info, not the catalog name?

Reply:
yea, since we won't support cross catalog renaming (it's not renaming anymore), I don't think it's necessary to contain catalog name in the `newNameParts`

Reply:
OK, that should simplify the logic. But, I have one concern. For the following:
```
ALTER TABLE testcat.ns1.ns2.old RENAME TO testcat.ns1.ns2.new
```
, `ns1.ns2.old` will be renamed to `testcat.ns1.ns2.new`. Is this the expected behavior or should we handle it gracefully?

Reply:
I think it is the expected behavior.